server/src/main/java/io/druid/metadata/SQLMetadataSegmentManager.java

```diff
@@ -312,8 +313,8 @@ public boolean removeDatasource(final String ds)
     try {
       final int removed = connector.getDBI().withHandle(
           handle -> handle.createStatement(
-              StringUtils.format("UPDATE %s SET used=false WHERE dataSource = :dataSource", getSegmentsTable())
-          ).bind("dataSource", ds).execute()
+              StringUtils.format("UPDATE %s SET used=false, used_update_date = :now WHERE dataSource = :dataSource", getSegmentsTable())
+          ).bind("dataSource", ds).bind("now", DateTimes.nowUtc().toString()).execute()
```

Comment:
minor nit: can we change update used_update_date even when someone enables the datasource ? 
this would not affect kill task, but would make the value in table consistent with the last time when someone toggled the used column

Reply:
made necessary updates to ensure new column staying up-to-date with segments creation, enable or disable.